google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/Connection.java

```diff
@@ -0,0 +1,718 @@
+   *       started, will cause a {@link SpannerException}
+   *   <li>Calling this method on a connection that already has a transaction that has started, will
+   *       cause a {@link SpannerException} (no nested transactions)
+   * </ul>
+   */
+  void beginTransaction();
+
+  /**
+   * Sets the transaction mode to use for current transaction. This method may only be called when
+   * in a transaction, and before the transaction is actually started, i.e. before any statements
+   * have been executed in the transaction.
+   *
+   * @param transactionMode The transaction mode to use for the current transaction.
+   *     <ul>
+   *       <li>{@link TransactionMode#READ_ONLY_TRANSACTION} will create a read-only transaction and
+   *           prevent any changes to written to the database through this transaction. The read
+   *           timestamp to be used will be determined based on the current readOnlyStaleness
+   *           setting of this connection. It is recommended to use {@link
+   *           TransactionMode#READ_ONLY_TRANSACTION} instead of {@link
+   *           TransactionMode#READ_WRITE_TRANSACTION} when possible, as read-only transactions do
+   *           not acquire locks on Cloud Spanner, and read-only transactions never abort.
+   *       <li>{@link TransactionMode#READ_WRITE_TRANSACTION} this value is only allowed when the
+   *           connection is not in read-only mode and will create a read-write transaction. If
+   *           {@link Connection#isRetryAbortsInternally()} is <code>true</code>, each read/write
+   *           transaction will keep track of a running SHA256 checksum for each {@link ResultSet}
+   *           that is returned in order to be able to retry the transaction in case the transaction
+   *           is aborted by Spanner.
+   *     </ul>
+   */
+  void setTransactionMode(TransactionMode transactionMode);
+
+  /**
+   * @return the transaction mode of the current transaction. This method may only be called when
+   *     the connection is in a transaction.
+   */
+  TransactionMode getTransactionMode();
+
+  /**
+   * @return <code>true</code> if this connection will automatically retry read/write transactions
+   *     that abort. This method may only be called when the connection is in read/write
+   *     transactional mode and no transaction has been started yet.
+   */
+  boolean isRetryAbortsInternally();
+
+  /**
+   * Sets whether this connection will internally retry read/write transactions that abort. The
+   * default is <code>true</code>. When internal retry is enabled, the {@link Connection} will keep
+   * track of a running SHA256 checksum of all {@link ResultSet}s that have been returned from Cloud
+   * Spanner. If the checksum that is calculated during an internal retry differs from the original
+   * checksum, the transaction will abort with an {@link
+   * AbortedDueToConcurrentModificationException}.
```

Comment:
No, I don't agree with that. Aborted transactions are handled by the connection API and could be relevant for any framework built on top of it. If the framework does not have some kind of automatic retry-loop for transactions, this feature of the connection API should be used to prevent the user from being confronted with aborted errors.
I've updated the comment to remove any references to JDBC, though.